sklearn/tree/_criterion.pxd

```diff
@@ -36,6 +38,9 @@ cdef class Criterion:
     cdef double weighted_n_node_samples  # Weighted number of samples in the node
     cdef double weighted_n_left          # Weighted number of samples in the left node
     cdef double weighted_n_right         # Weighted number of samples in the right node
+    cdef double weighted_n_missing       # Weighted number of samples that are missing
+
+    cdef public bint supports_missing_values  # True if Criterion supports missing values
```

Comment:
This attribute doesn't seem to be used. Should it be used to separate `_move_sums()` into one function with missing-value support, and one for the regular implementation?

Reply:
This was an experiment and was not suppose to be on this branch. I was coming up with an API to allow third party Criterions to support missing values. Overall, I want to get missing value support into scikit-learn first. After that, we can think about enabling third party Criterion's to support missing values.

Reply:
Yeah sounds good! Just reviewing Cython PRs to help.